### PR TITLE
Update Rigid3d Machine Definitions

### PR DESCRIPTION
**Pull request by mehmetsutas**
Small updates made on Rigid3D printers base and mucit2 definition files.

Added definition file for new model Zero3.

### CODE REVIEW

File: resources/definitions/rigid3d_base.def.json [hunk 1]
```diff
@@ -18,96 +18,107 @@
...
-            "ultimaker_bam",
-            "ultimaker_cpe_black", "ultimaker_cpe_blue", "ultimaker_cpe_dark-grey", "ultimaker_cpe_green", "ultimaker_cpe_light-grey", "ultimaker_cpe_plus_black", "ultimaker_cpe_plus_transparent", "ultimaker_cpe_plus_white", "ultimaker_cpe_red", "ultimaker_cpe_transparent", "ultimaker_cpe_white", "ultimaker_cpe_yellow",
-            "ultimaker_nylon_black", "ultimaker_nylon_transparent",
-            "ultimaker_pc_black", "ultimaker_pc_transparent", "ultimaker_pc_white",
-            "ultimaker_pla_black", "ultimaker_pla_blue", "ultimaker_pla_green", "ultimaker_pla_magenta", "ultimaker_pla_orange", "ultimaker_pla_pearl-white", "ultimaker_pla_red", "ultimaker_pla_silver-metallic", "ultimaker_pla_transparent", "ultimaker_pla_white", "ultimaker_pla_yellow",
-            "ultimaker_pp_transparent",
-            "ultimaker_pva",
-            "ultimaker_tough_pla_black", "ultimaker_tough_pla_green", "ultimaker_tough_pla_red", "ultimaker_tough_pla_white",
-            "ultimaker_tpu_black", "ultimaker_tpu_blue", "ultimaker_tpu_red", "ultimaker_tpu_white",
+            "ultimaker_abs_black", "ultimaker_abs_blue", "ultimaker_abs_green", "ultimaker_abs_grey",
```
Comment (nallath) [inline resources/definitions/rigid3d_base.def.json:52]: The ultimaker profiles don't even need to be mentioned here, as this is a 1.75mm printer. They are filtered out automatically (DRY principle; Don't repeat yourself)

**Review by nallath — CHANGES_REQUESTED** [review]

File: resources/definitions/rigid3d_base.def.json (hunk 1 above)
Comment (mehmetsutas) [reply resources/definitions/rigid3d_base.def.json]: Didn't knew that. Removed the material exclusions. Thank you.

**Comment by nallath** [comment]
It seems there are some conflicts with the main branch. Could you have a look at them?

**Comment by mehmetsutas** [comment]
@nallath solved the conflicts. Excuse me for missing it at first hand.

File: resources/quality/rigid3d_base/nylon/rigid3d_base_nylon_good.inst.cfg [hunk 2]
```diff
@@ -10,8 +10,8 @@ quality_type = good
 material = generic_nylon
 
 [values]
-default_material_print_temperature = 245
-default_material_bed_temperature = 90
+default_material_print_temperature = 250
+default_material_bed_temperature = 100
```
Comment (nallath) [inline resources/quality/rigid3d_base/nylon/rigid3d_base_nylon_good.inst.cfg:14]: Sorry! I missed this. 

We used to accept this in the past, but that's not correct. Basically you shouldn't set default_material_print_temperature or default_material_bed_temperature in quality profiles. The reason for this is that the materials (that are below the quality profiles) set this. If you want a certain material_print_temperature that is higher (so say that the default is set by the material to 240) you should use material_bed_temperature = =default_material_print_temperature + 10

Comment (mehmetsutas) [reply resources/quality/rigid3d_base/nylon/rigid3d_base_nylon_good.inst.cfg]: @nallath corrections are done as per your suggestion.

**Comment by jellespijker** [comment]
Can you sync your fork, with our `main`. We have added a **printer-linter** tool for our printer definitions and profiles. This should help us with processing our PR's. It performs some standard diagnostic checks and apply auto-formatting on the changed files in this PR, but you can also run it locally. See https://github.com/Ultimaker/Cura/tree/main/printer-linter

If it finds issues for which it knows a fix it should post a comment with that suggested fix such that you can easily commit that fix if you're okay with it.

File: resources/definitions/rigid3d_base.def.json [hunk 3]
```diff
@@ -3,101 +3,66 @@
...
+        "infill_pattern": { "value": "'zigzag'" },
 
         "material_initial_print_temperature": { "value": "material_print_temperature" },
         "material_final_print_temperature": { "value": "material_print_temperature" },
         "material_bed_temperature_layer_0": { "value": "material_bed_temperature" },
 
-        "wall_0_material_flow": { "value": 95 },
-
-        "speed_print": { "value": 40.0 } ,
+       } ,
```
Comment (jellespijker) [inline resources/definitions/rigid3d_base.def.json]:
```suggestion
```

File: resources/definitions/rigid3d_base.def.json [hunk 4]
```diff
@@ -106,14 +71,17 @@
...
-        "retraction_amount": { "value": 1.50 },
-        "retraction_prime_speed": { "value": 7 },
-        "retraction_count_max": { "value": 5 },
-        "retraction_combing": { "value": "'noskin'" },
-        "travel_avoid_other_parts": { "value": "False" },
-        
-        "support_z_distance": { "value": "layer_height if layer_height >= 0.16 else layer_height * 2" },
-        "support_interface_enable": { "value": true },
+        "retraction_amount": { "value": 1.5 },
+       ,
```
Comment (jellespijker) [inline resources/definitions/rigid3d_base.def.json]:
```suggestion
````

File: resources/definitions/rigid3d_base.def.json [hunk 5]
```diff
@@ -106,14 +71,17 @@
...
+        "travel_retract_before_outer_wall": { "value": "True" },
+        "retraction_combing": { "value": "'all'" },
+        "retraction_combing_max_distance": { "value": 30 },
+
+       ,
```
Comment (jellespijker) [inline resources/definitions/rigid3d_base.def.json]:
```suggestion
```

**Review by jellespijker — CHANGES_REQUESTED** [review]
I added some code suggestions to fix some orphaned `,` and `}`

If you check the files tab in this PR you can add multiple code suggestion to the batch and process them as a singe comment

File: resources/definitions/rigid3d_base.def.json (hunk 5 above)
Comment (mehmetsutas) [reply resources/definitions/rigid3d_base.def.json]: Excuse me for the inconvenience. Just approved the requested changes by the bot and  things mixed up. Corrected the file and it should be good now.

Comment (mehmetsutas) [reply resources/definitions/rigid3d_base.def.json]: @jellespijker This PR has waited a lot. Is there chance of it being it included in the next release?

**Pull request closed without merging**